### DISCUSSION TITLE
Solar Generation Total

### DISCUSSION THREAD

**Discussion by Captain0Kangaroo** [nathanmarlor/foxess_modbus#134] [General]
I've been using the Solar Generation Total (sensor.solar_energy_total) which until yesterday had a value of about 6,553kwh which seemed about right and i'm guessing this figure was pulled from the inverter. 
Today when i had a look to see what the generation was up to it was only reading 29.8kwh.

Any reason why this data has been reset/ lost?

![Untitled 2](https://user-images.githubusercontent.com/96144685/230737086-f70b1169-76c1-45a5-9468-886355ece2e1.jpg)

**Comment by canton7** [comment]
Aha, yes. Would you mind reading register 11069, it should be 1.

If so, this is fixed in the next release.

**Reply by Captain0Kangaroo** [reply]
I've no idea how to read the registers.

**Reply by canton7** [reply]
Download https://www.modbusdriver.com/modpoll.html#:~:text=modpoll%20is%20a%20command%20line,Linux%20PCs then `c:\path-to-modpoll\modpoll.exe -m tcp -p 502 -a 247 -t 3 -0 -1 -c 8 -r 11069 192.x.x.x.x` (where 192.x.x.x is the ip address of your ethernet to RS485 adapter).

Alternatively, I imagine the next beta will be out soon.

**Reply by Captain0Kangaroo** [reply]
I'm Mac so can't use it unfortunately.

**Reply by Captain0Kangaroo** [reply]
> Alternatively, I imagine the next beta will be out soon.


With the next beta do you think this will restore the total back to what it should be or will it just continue from where it is?

**Reply by canton7** [reply]
It will restore it back.

The problem is that we're effectively missing the higher digits: conceptually we went from 999 to 1000, but because we're not reading the 4th digit we show 0 instead of 1000. (Although we rolled over from 6,553.5, because binary).

**Reply by Captain0Kangaroo** [reply]
Great. Thanks 👍

**Reply by canton7** [reply]
This should now be fixed

**Reply by Captain0Kangaroo** [reply]
Yep.. Working fine now.
